src/core/utils/ping_sender.cpp

```diff
@@ -180,12 +210,25 @@ void PingSender::HandleIcmpReceive(const Message &          aMessage,
     reply.mSequenceNumber = aIcmpHeader.GetSequence();
     reply.mHopLimit       = aMessageInfo.GetHopLimit();
 
+    VerifyOrExit(reply.mRoundTripTime <= mConfig.mTimeout);
+
+    mStatistics.mReceivedCount++;
+    mStatistics.mTotalRoundTripTime += reply.mRoundTripTime;
+    mStatistics.mMaxRoundTripTime = OT_MAX(mStatistics.mMaxRoundTripTime, reply.mRoundTripTime);
+    mStatistics.mMinRoundTripTime = OT_MIN(mStatistics.mMinRoundTripTime, reply.mRoundTripTime);
+
+    mConfig.InvokeReplyCallback(reply);
+    // Received all ping replies, no need to wait longer.
+    if (aIcmpHeader.GetSequence() == mTargetEchoSequence)
```

Comment:
This is not strictly correct. 

Besides `PingSender`, user can also use `otIcmp6SendEchoRequest` to send `Ping Requets` even when the `Ping Sender` is busy. `otIcmp6SendEchoRequest` will increase the `Echo Sequence` so that `mTargetEchoSequence` won't match the last echo sequence of `Ping Requests` sent by `Ping Sender`.

I think we should set `mTargetEchoSequence` to the next Echo Sequence in `SendPing` instead of setting it once in `Ping()`, so that `mTargetEchoSequence` effectively tracks the next Echo Sequence to expect, as below:

```suggestion
    if (mCount == 0 && aIcmpHeader.GetSequence() == mTargetEchoSequence)
```

Thoughts?